src/content/get-started/async.md

```diff
@@ -0,0 +1,374 @@
+* Run the application to see HTTP data fetching in action.
+
+:::
+
+## Prerequisites
+
+* Completion of Chapter 2, which covered basic Dart syntax and command-line
+  interaction. You should have a `dartpedia` project set up.
+* Familiarity with the concept of APIs (Application Programming Interfaces) as a
+  way to retrieve data.
+
+## Tasks
+
+In this chapter, you will modify the existing `dartpedia` CLI application to fetch and
+display an **article summary** using the `http` package and asynchronous
+programming techniques.
+
+### Task 1: Adding the http dependency
+
+Before you can make HTTP requests, you need to add the `http` package
+as a dependency to your project.
+
+1.  Open the `dartpedia/pubspec.yaml` file within your project. This file is
+    called the **pubspec**, and it manages your Dart project's metadata,
+    dependencies (like the `http` package), and assets.
+2.  Locate the `dependencies` section.
+3.  Add `http: ^1.3.0` (or the latest stable version) under `dependencies`. The
+    `^` symbol allows compatible versions to be used.
+
+    ```yaml
+    dependencies:
+      http: ^1.3.0
+    ```
+
+4.  Save the `pubspec.yaml` file.
+5.  Run `dart pub get` in your terminal from the `dartpedia` directory. This
+    command fetches the newly added dependency and makes it available for
+    use in your project.
+
+    You should see output similar to this:
+
+    ```bash
+    Resolving dependencies...  
+    Downloading packages...  
+    + http 1.4.0
+      lints 5.1.1 (6.0.0 available)
+    Changed 1 dependency!
+    1 package has newer versions incompatible with dependency constraints.
+    Try `dart pub outdated` for more information.
+    ```
+
+### Task 2: Importing the http package
+
+Now that you've added the `http` package, you need to import it into
+your Dart file to use its functionalities.
+
+1.  Open the `dartpedia/bin/cli.dart` file.
+2.  Add the following `import` statement at the top of the file, along
+    with the existing `dart:io` import:
+
+    ```dart
+    import 'dart:io';
+    import 'package:http/http.dart' as http; // Add this line
+    ```
+
+    This line imports the `http` package and gives it the alias `http`.
+    This allows you to refer to classes and functions within the `http`
+    package using `http.` (e.g., `http.Client`, `http.get`). The `as http`
+    part is a standard convention to avoid naming conflicts if another
+    imported library also has a similarly named class or function.
+
+### Task 3: Implement the `getWikipediaArticle` function
+
+You will now create a new function called `getWikipediaArticle` that
+handles fetching data from an external API. This function will be `async`
+because network requests are asynchronous operations.
+
+1.  **Define the function signature:**
+    Below your `main` function (and `printUsage` function), add the
+    following function signature.
+    The `Future<String>` return type indicates that this function will
```

Comment:
I would turn this into a bulleted list that comes below the example function signature. This is to match the structure in Chapter 1 & 2. So...

```dart
function signature...
```

Highlights from the preceding code:

* `Future<String>` does a thing.
* `async` does a thing.